### PR TITLE
convert Expr.Typ to value

### PR DESCRIPTION
**Pull request by Morranto**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #https://github.com/matrixorigin/MO-Cloud/issues/2451
## What this PR does / why we need it:
reduce the usage of object memory ref

### CODE REVIEW

**Comment by aunjgr** [comment]
should have benchmark comparison on tp workloads

File: pkg/sql/plan/apply_indices.go [hunk 1]
```diff
@@ -319,7 +319,7 @@ END0:
 			}
 			idxFilter, _ = BindFuncExprImplByPlanExpr(builder.GetContext(), funcName, []*plan.Expr{
 				{
-					Typ: DeepCopyType(idxTableDef.Cols[0].Typ),
+					Typ: *DeepCopyType(idxTableDef.Cols[0].Typ),
```
Comment (fengttt) [inline pkg/sql/plan/apply_indices.go:322]: This one, Typ: *DeepCopyType(XXX),          isn't this just Typ: XXX?

**Review by fengttt — CHANGES_REQUESTED** [review]

**Review by ouyuanning — APPROVED** [review]

File: pkg/sql/plan/apply_indices.go (hunk 1 above)
Comment (Morranto) [reply pkg/sql/plan/apply_indices.go]: done

**Comment by Morranto** [comment]
sysbench update index result:

before pr:
image
after:
image

**Comment by Morranto** [comment]
> should have benchmark comparison on tp workloads

has been updated

**Review by nnsgmsone — APPROVED** [review]

**Review by m-schen — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Review by badboynt1 — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Comment by Morranto** [comment]
please review this pr @fengttt

**Review by fengttt — APPROVED** [review]

**Pull request merged**